pkg/nsx/services/inventory/service.go

```diff
@@ -32,3 +36,41 @@ func (s *InventoryService) initContainerApplication(clusterId string) error {
 	}
 	return nil
 }
+
+func (s *InventoryService) SyncContainerApplication(name, namespace string, key InventoryKey) *InventoryKey {
+	service := &corev1.Service{}
+	err := s.Client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, service)
+	externalId := key.ExternalId
+	if apierrors.IsNotFound(err) ||
+		((err == nil) && (string(service.UID) != externalId)) {
+		err = s.DeleteResource(externalId, ContainerApplication)
+		if err != nil {
+			log.Error(err, "Delete ContainerApplication Resource error", "key", key)
+			return &key
+		}
+	} else if err == nil {
+		retry := s.BuildService(service)
+		if retry {
+			return &key
+		}
+	} else {
+		log.Error(err, "Unexpected error is found while processing Service")
+	}
+	return nil
+}
+
+func (s *InventoryService) DeleteContainerApplication(externalId string, inventoryObject *containerinventory.ContainerApplication) error {
```

Comment:
So do we need to update containerApplicationIds in pod for service deletion in this function?

Reply:
@TaoZou1 would consider it in pod logic